crd/external/podnetworkinstance/api/v1alpha1/podnetworkinstance.go

```diff
@@ -0,0 +1,59 @@
+//go:build !ignore_uncovered
+// +build !ignore_uncovered
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Important: Run "make" to regenerate code after modifying this file
+
+// +kubebuilder:object:root=true
+
+// PodNetworkInstance is the Schema for the PodNetworkInstances API
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:resource:shortName=pni
+// +kubebuilder:subresource:status
+// +kubebuilder:metadata:labels=managed=
+// +kubebuilder:metadata:labels=owner=
+// +kubebuilder:printcolumn:name="Pod IPs",type=string,priority=1,JSONPath=`.status.podIPAddresses`
+// +kubebuilder:printcolumn:name="PodNetwork",type=string,priority=1,JSONPath=`.spec.podNetwork`
+// +kubebuilder:printcolumn:name="PodIPReservationSize",type=string,priority=1,JSONPath=`.spec.podIPReservationSize`
+type PodNetworkInstance struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   PodNetworkInstanceSpec   `json:"spec,omitempty"`
+	Status PodNetworkInstanceStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// PodNetworkInstanceList contains a list of PodNetworkInstance
+type PodNetworkInstanceList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []PodNetworkInstance `json:"items"`
+}
+
+// PodNetworkInstanceSpec defines the desired state of PodNetworkInstance
+type PodNetworkInstanceSpec struct {
+	// +kubebuilder:default=0
+	// +kubebuilder:validation:Optional
```

Comment:
not exactly sure what this annotation means but a PodNetwork is not an optional field of a PodNetworkInstance

Reply:
hmm, just removing this annotation is not enough to make this field required. I would also need to drop omit empty from the field for it to be marked required in CRD
With omitempty in place this annotation has no effect.

Reply:
so I dropped omitempty from podnetwork and it is now marked req in generated CRD.